test/common/test_context.go

```diff
@@ -659,7 +659,18 @@ func (ctx *TestContext) RegisterBrokerWithCatalogAndLabelsExpect(catalog SBCatal
 	}
 
 	MergeObjects(brokerJSON, brokerData)
-	broker := RegisterBrokerInSM(brokerJSON, expect, map[string]string{})
+	response := RegisterBrokerInSMWithoutErrorSwallowing(brokerJSON, expect, map[string]string{})
+	responseStatusCode := response.Raw().StatusCode
+	if http.StatusCreated != responseStatusCode {
```

Comment:
IMHO, this is needed since in registerBrokerWithCatalog function (notifications_suite-test.go), we call GetBrokerAsParams() on the retrieved BrokerUtils object.
Hadn't we had this error handling, we would've needed to check whether BrokerUtils isn't nil and return a null value for 'id' and 'BrokerServer' elswise!
Hence, I think that having this error handling logic is more allegiant and clearer than the aforementioned alternative.